### PR TITLE
Fix: removed unneeded HTML escaping

### PR DESCRIPTION
**Pull request by annando**
The escaping had been unnecessary. I also had a rough look where used the functions to create HTML entities and reverse them. It seems as if (at least for the body) this isn't done at any place now anymore.

### CODE REVIEW

File: src/Content/Text/HTML.php [hunk 1]
```diff
@@ -336,8 +336,6 @@ function ($matches) use (&$codeblocks) {
 
 		$message = html_entity_decode($message, ENT_QUOTES, 'UTF-8');
 
-		$message = str_replace(["<"], ["&lt;"], $message);
```
Comment (nupplaphil) [inline src/Content/Text/HTML.php]: can you add two files here:
https://github.com/friendica/friendica/tree/develop/tests/datasets/content/text/html

bug-8075.html **with** "<" text
bug-8075.txt with the encoded, expected content?

thx :)

Comment (annando) [reply src/Content/Text/HTML.php]: Seems like I can't.

Comment (nupplaphil) [reply src/Content/Text/HTML.php]: This is quite interesting because you did it the right way .. the "<>" text should **not** be removed, but the tests says it will be removed see the result here: https://travis-ci.org/friendica/friendica/jobs/633997421#L421-L426

So are you sure "<>" won't get escaped anymore? @MrPetovan am I missing something for the test here?
https://github.com/friendica/friendica/blob/5cc2dc7ca37f5037bbba7d2fa8cb408d0b37c250/tests/src/Content/Text/HTMLTest.php#L73-L79

Comment (nupplaphil) [reply src/Content/Text/HTML.php]: @annando my fault, I took the wrong testcase!

Under
https://github.com/friendica/friendica/blob/5cc2dc7ca37f5037bbba7d2fa8cb408d0b37c250/tests/src/Content/Text/HTMLTest.php#L57-L60
add these new array content:

```php
			'bug-8075-html-tags => [
				'expectedBBCode' => '<big rant here> I don't understand tests',
				'html' => '<big rant here> I don't understand tests',
			],
```

This will get automatically passed to this test:
https://github.com/friendica/friendica/blob/5cc2dc7ca37f5037bbba7d2fa8cb408d0b37c250/tests/src/Content/Text/HTMLTest.php#L73-L78
which checks the expected result :-)

Comment (nupplaphil) [reply src/Content/Text/HTML.php]: Ok forget it ^^ .. I think I'm currently too stupid to guide you well enough..

**Comment by annando** [comment]
Gnarf, the last commit included some stuff that I didn't intended to commit now ...

**Comment by annando** [comment]
Hmm, the tests behave weird ...

**Comment by MrPetovan** [comment]
Nope, since angle brackets aren't escaped in the source HTML, the `strip_tags` does its job. The HTML source should be `&lt;big rant here&gt; I don&quot;t understand tests`

File: tests/src/Content/Text/HTMLTest.php [hunk 2]
```diff
@@ -58,6 +58,10 @@ public function dataHTMLText()
 				'expectedBBCode' => '[audio]http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3[/audio]',
 				'html' => '<audio src="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3" controls="controls"><a href="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3">http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3</a></audio>',
 			],
+			'bug-8075-html-tags' => [
+				'expectedBBCode' => "<rant here> I don't understand tests",
+				'html' => "<rant here> I don't understand tests",
```
Comment (MrPetovan) [inline tests/src/Content/Text/HTMLTest.php]:
```suggestion
				'html' => '&lt;rant here&gt; I don&quot;t understand tests',
```

Comment (annando) [reply tests/src/Content/Text/HTMLTest.php]: The text arrives without the entities. The input is okay. So the test is failing. We mustn't change the input to match the output, but we have to possibly change the test to do the right stuff.

Comment (MrPetovan) [reply tests/src/Content/Text/HTMLTest.php]: If the text arrives without the entities when it is expected to be HTML, then we have to assume they are regular tags and `strip_tags` is doing its job.

We can have an additional test case like this:
```php
			'bug-8075-html-tags-unescaped' => [
				'expectedBBCode' => "I don't understand tests",
				'html' => "<rant here> I don't understand tests",
			],
```

And both tests would have the correct expected output. We just can't assume literal angle brackets in incoming HTML messages aren't tags.

Comment (MrPetovan) [reply tests/src/Content/Text/HTMLTest.php]: Remember tests don't care about the context, so theoretically we have to cover every possible context and provide the expected output for each. In practice we only cover the corner cases we encounter when fixing bugs because we already test obvious cases during development.

Comment (annando) [reply tests/src/Content/Text/HTMLTest.php]: Yeah, I had taken a wrong turn during my thinking about this :grin:

File: tests/src/Content/Text/HTMLTest.php [hunk 3]
```diff
@@ -58,6 +58,10 @@ public function dataHTMLText()
 				'expectedBBCode' => '[audio]http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3[/audio]',
 				'html' => '<audio src="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3" controls="controls"><a href="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3">http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3</a></audio>',
 			],
+			'bug-8075-html-tags' => [
+				'expectedBBCode' => "&lt;big rant here&rt; I don't understand tests",
+				'html' => "<big rant here> I don't understand tests",
```
Comment (MrPetovan) [inline tests/src/Content/Text/HTMLTest.php]: No, it's the opposite, the `html` property should have the entities, the `expectedBBCode` shouldn't!

Comment (annando) [reply tests/src/Content/Text/HTMLTest.php]: Gnarf ...

**Comment by MrPetovan** [comment]
Test is failing:
```
1) Friendica\Test\src\Content\Text\HTMLTest::testToBBCode with data set "bug-8075-html-tags" ('<big rant here> I don't under... tests', '&lt;big rant here&rt; I don't... tests')
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'<big rant here> I don't understand tests'
+'<big rant here&rt; I don't understand tests'
```

Now there seem to be an issue with right angle bracket.

File: tests/src/Content/Text/HTMLTest.php [hunk 4]
```diff
@@ -58,6 +58,10 @@ public function dataHTMLText()
 				'expectedBBCode' => '[audio]http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3[/audio]',
 				'html' => '<audio src="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3" controls="controls"><a href="http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3">http://www.cendrones.fr/colloque2017/jonathanbocquet.mp3</a></audio>',
 			],
+			'bug-8075-html-tags' => [
+				'expectedBBCode' => "<big rant here> I don't understand tests",
+				'html' => "&lt;big rant here&rt; I don't understand tests",
```
Comment (MrPetovan) [inline tests/src/Content/Text/HTMLTest.php]:
```suggestion
				'html' => "&lt;big rant here&gt; I don't understand tests",
```
Not `&rt;`. `lt` stands for `less than`, not `LefT` which would warrant a `rt` as `RighT`.

Comment (annando) [reply tests/src/Content/Text/HTMLTest.php]: Yeah, I guess I shouldn't watch Youtube videos while coding.
